### PR TITLE
feat(cli): Tag Support

### PR DESCRIPTION
**Pull request by sebastiancrossa**
*Issue #, if available:*
#4607 

*Description of changes:*
This PR implements tags into the Amplify CLI workflow. I'd love to receive as much feedback! There are definitely things I know that can be done better. Feel free to let me know if there are parts of the code you don't understand too.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-cli/src/init-steps/s9-onSuccess.js [hunk 1]
```diff
@@ -141,6 +153,31 @@ function generateGitIgnoreFile(context) {
...
+  let returnedArr = arr;
+
+  returnedArr.forEach(tagObj => {
+    if (
+      tagObj['Value'].includes('{project-env}') ||
+      tagObj['Value'].includes('{project-name') ||
+      tagObj['Value'].includes('{cli-version')
+    ) {
+      const replaceWith = {
+        '{project-name}': 'foobar',
```
Comment (sebastiancrossa) [inline packages/amplify-cli/src/init-steps/s9-onSuccess.js:166]: This function isn't being used and was initially added for testing. I forgot to delete this part of the code before pushing, but that will be taken care of on the next commit pushed.

**Comment by ammarkarachi** [comment]
@sebastiancrossa Can you fix the conflicts?

File: packages/amplify-cli/src/init-steps/s9-onSuccess.js [hunk 2]
```diff
@@ -141,6 +153,31 @@ function generateGitIgnoreFile(context) {
...
+function replaceTags(arr) {
+  let returnedArr = arr;
+
+  returnedArr.forEach(tagObj => {
+    if (
+      tagObj['Value'].includes('{project-env}') ||
+      tagObj['Value'].includes('{project-name') ||
+      tagObj['Value'].includes('{cli-version')
+    ) {
+      const replaceWith = {
```
[the remark this answers is not in the document]
Comment (sebastiancrossa) [reply packages/amplify-cli/src/init-steps/s9-onSuccess.js:165]: Nope! I mentioned it on a review I left a couple of days ago, I completely forgot to remove that part of the code :p

File: packages/amplify-provider-awscloudformation/src/aws-utils/aws-cfn.js [hunk 3]
```diff
@@ -188,6 +188,23 @@ class CloudFormation {
       });
   }
 
+  replaceTagVars(tagsArr) {
```
Comment (ammarkarachi) [inline packages/amplify-provider-awscloudformation/src/aws-utils/aws-cfn.js:191]: Why is this here? Is it not possible to put this functionality where all the other tag functionality

File: packages/amplify-provider-awscloudformation/lib/push-resources.js [hunk 4]
```diff
@@ -41,7 +41,7 @@ async function run(context, resourceDefinition) {
 
     validateCfnTemplates(context, resources);
 
-    await packageResources(context, resources);
+    await packageResources(context, resources, projectDetails);
```
Comment (ammarkarachi) [inline packages/amplify-provider-awscloudformation/lib/push-resources.js:44]: What is projectDetails?

File: packages/amplify-e2e-core/src/utils/projectMeta.ts [hunk 5]
```diff
@@ -31,6 +31,17 @@ function getBackendAmplifyMeta(projectRoot: string) {
...
 }
 
+function getProjectTags(projectRoot: string) {
+  // Makes sure that the tags file is a valid JSON
+  // ? Not sure if this is the best way to go, since I'm not seeing the other files do something similar for error handling
+  try {
+    const tagsFilePath = path.join(projectRoot, 'amplify', 'tags.json');
+    return JSON.parse(fs.readFileSync(tagsFilePath, 'utf8'));
+  } catch (err) {
+    throw new Error(err);
```
Comment (ammarkarachi) [inline packages/amplify-e2e-core/src/utils/projectMeta.ts:41]: Where is this error handler?

File: packages/amplify-cli/src/extensions/amplify-helpers/resource-status.js [hunk 6]
```diff
@@ -9,6 +9,7 @@ const _ = require('lodash');
 const { CLOUD_INITIALIZED, CLOUD_NOT_INITIALIZED, getCloudInitStatus } = require('./get-cloud-init-status');
 const { readJsonFile } = require('./read-json-file');
 const { ServiceName: FunctionServiceName, hashLayerResource } = require('amplify-category-function');
+const { CloudFormation } = require('aws-sdk');
```
Comment (ammarkarachi) [inline packages/amplify-cli/src/extensions/amplify-helpers/resource-status.js:12]: Use `aws-provider-cloudformation`

**Comment by ammarkarachi** [comment]
Can you fix the conflicts?

File: packages/amplify-provider-awscloudformation/lib/push-resources.js (hunk 4 above)
Comment (sebastiancrossa) [reply packages/amplify-provider-awscloudformation/lib/push-resources.js:44]: I had it for a previous implementation that is now refactored, so it's not needed anymore. This will be updated on the next push!

File: packages/amplify-cli/src/extensions/amplify-helpers/resource-status.js (hunk 6 above)
Comment (sebastiancrossa) [reply packages/amplify-cli/src/extensions/amplify-helpers/resource-status.js:12]: It seems like CloudFormation from the `aws-provider-cloudformation` library can't be directly replaced without making other changes (getting a "CloudFormation is not a constructor" error).

Is there anything else I'd need to modify?

Comment (ammarkarachi) [reply packages/amplify-cli/src/extensions/amplify-helpers/resource-status.js:12]: remove this for now you cannot use a non configure aws-sdk directly
